src/js/components/search/visualizations/geo/MapLegendHeader.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { TooltipWrapper } from 'data-transparency-ui';
+import { mapLegendTT } from './MapLegendTooltip';
+
+const propTypes = {
+    mapLegendToggleData: PropTypes.arrayOf(PropTypes.shape({
+        title: PropTypes.string,
+        value: PropTypes.string
+    })),
+    mapLegendToggle: PropTypes.string,
+    updateToggle: PropTypes.func
+};
+
+const MapLegendHeader = ({ mapLegendToggleData, mapLegendToggle, updateToggle }) => {
+    const headerToggle = () => {
+        if (!mapLegendToggleData) return null;
+
+        return (mapLegendToggleData?.map((toggleButtonData) => (
```

Comment:
More so for my own curiosity, but does the null operator here actually matter since you do a falsy check above?

Reply:
correct

Reply:
this is optional chaining https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

null coalescing operater is this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator